File: androidApp/src/main/kotlin/com/malaria/android/ui/onboarding/admin/ClinicDetailsStep.kt
package com.malaria.android.ui.onboarding.admin

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.malaria.android.ui.locals.LocalOnboardingState
import com.malaria.android.ui.onboarding.components.JurisdictionPicker
import com.malaria.android.ui.onboarding.components.LawfulBasisPicker
import com.malaria.android.ui.onboarding.components.WizardStepContainer
import com.malaria.domain.Jurisdiction
import com.malaria.domain.LawfulBasis
import kotlinx.coroutines.launch

/**
 * Step 5 of admin provisioning (spec §10): clinic name + jurisdiction +
 * lawful basis.
 *
 * Mirrors `iosApp/Views/Onboarding/Admin/ClinicDetailsStep.swift`.
 */
@Composable
fun ClinicDetailsStep() {
    val onboarding = LocalOnboardingState.current
    val scope = rememberCoroutineScope()

    var clinicName by remember { mutableStateOf("") }
    var jurisdiction by remember { mutableStateOf(Jurisdiction.OTHER) }
    var lawfulBasis by remember { mutableStateOf(LawfulBasis.HEALTH_PROVISION) }

    val trimmed = clinicName.trim()

    WizardStepContainer(
        title = "Clinic details",
        subtitle = "These appear on exports and in the audit log. You can change them later in Settings (admin biometric required).",
        stepIndicator = "Step 5 of 8",
        primaryLabel = "Continue",
        primaryEnabled = trimmed.isNotEmpty(),
        onPrimary = {
            scope.launch {
                onboarding.configureClinic(
                    name = trimmed,
                    jurisdiction = jurisdiction.canonical,
                    lawfulBasis = lawfulBasis.canonical,
                    version = "v1.0",
                )
            }
        },
    ) {
        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            Text("Clinic name", style = MaterialTheme.typography.titleSmall)
            OutlinedTextField(
                value = clinicName,
                onValueChange = { clinicName = it },
                placeholder = { Text("e.g. Kisumu District Health Centre") },
                singleLine = true,
                modifier = Modifier.fillMaxWidth(),
            )
        }

        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            Text("Jurisdiction", style = MaterialTheme.typography.titleSmall)
            JurisdictionPicker(
                selection = jurisdiction,
                onSelectionChange = { jurisdiction = it },
            )
        }

        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            Text("Lawful basis", style = MaterialTheme.typography.titleSmall)
            LawfulBasisPicker(
                selection = lawfulBasis,
                onSelectionChange = { lawfulBasis = it },
            )
        }
    }
}

File: iosApp/Views/Onboarding/Admin/ClinicDetailsStep.swift
import SwiftUI
@preconcurrency import Shared

/// Step 5 of admin provisioning (spec §10): clinic name + jurisdiction +
/// lawful basis.
struct ClinicDetailsStep: View {

    @Environment(OnboardingState.self) private var onboarding

    @State private var clinicName: String = ""
    @State private var jurisdiction: Jurisdiction = .other
    @State private var lawfulBasis: LawfulBasis = .healthProvision
    @State private var errorMessage: String?

    private var trimmedName: String {
        clinicName.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    var body: some View {
        WizardStepContainer(
            title: "Clinic details",
            subtitle: "These appear on exports and in the audit log. You can change them later in Settings (admin biometric required).",
            stepIndicator: "Step 5 of 8",
            primary: .init(title: "Continue", isEnabled: !trimmedName.isEmpty) {
                do {
                    try onboarding.configureClinic(
                        name: trimmedName,
                        jurisdiction: jurisdiction.canonical,
                        lawfulBasis: lawfulBasis.canonical,
                        version: "v1.0"
                    )
                } catch {
                    errorMessage = error.localizedDescription
                }
            }
        ) {
            VStack(alignment: .leading, spacing: 8) {
                Text("Clinic name")
                    .font(.headline)
                TextField("e.g. Kisumu District Health Centre", text: $clinicName)
                    .textFieldStyle(.roundedBorder)
                    .autocorrectionDisabled(false)
                    .submitLabel(.done)
            }

            VStack(alignment: .leading, spacing: 8) {
                Text("Jurisdiction")
                    .font(.headline)
                JurisdictionPicker(selection: $jurisdiction)
            }

            VStack(alignment: .leading, spacing: 8) {
                Text("Lawful basis")
                    .font(.headline)
                LawfulBasisPicker(selection: $lawfulBasis)
            }
        }
        .alert(
            "Unable to save clinic details",
            isPresented: Binding(
                get: { errorMessage != nil },
                set: { if !$0 { errorMessage = nil } }
            )
        ) {
            Button("OK", role: .cancel) { errorMessage = nil }
        } message: {
            Text(errorMessage ?? "")
        }
    }
}
